File: python3/deepest_leaves_sum.py
from .model import TreeNode

"""
Space : O(n)
Time  : O(n)
"""

class Solution:
    def deepestLeavesSum(self, root: TreeNode) -> int:        
        stack = [root]
        
        while stack:
            temp = []
            total = 0
            for i in stack:
                total += i.val
                if i.left:
                    temp.append(i.left)
                if i.right:
                    temp.append(i.right)
            if not temp:
                return total
            stack = temp
            
        return 0

File: golang/deepest_leaves_sum.go
package golang

import (
	"golang/model"
)

func deepestLeavesSum(root *model.TreeNode) int {
	stack := []*model.TreeNode{root}

	for len(stack) > 0 {
		temp := []*model.TreeNode{}
		ans := 0
		for _, node := range stack {
			if node.Left != nil {
				temp = append(temp, node.Left)
			}
			if node.Right != nil {
				temp = append(temp, node.Right)
			}
			ans += node.Val
		}

		if len(temp) == 0 {
			return ans
		}

		stack = temp
	}

	return 0
}
